jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java

```diff
@@ -143,33 +150,25 @@ public CredentialRetriever dockerCredentialHelper(Path credentialHelper) {
   }
 
   /**
-   * Creates a new {@link CredentialRetriever} that tries common Docker credential helpers to
+   * Creates a new {@link CredentialRetriever} that tries well-known Docker credential helpers to
    * retrieve credentials based on the registry of the image, such as {@code docker-credential-gcr}
-   * for images with the registry as {@code gcr.io}.
+   * for images with the registry ending with {@code gcr.io}.
    *
    * @return a new {@link CredentialRetriever}
    */
-  public CredentialRetriever inferCredentialHelper() {
-    List<String> inferredCredentialHelperSuffixes = new ArrayList<>();
-    for (String registrySuffix : COMMON_CREDENTIAL_HELPERS.keySet()) {
-      if (!imageReference.getRegistry().endsWith(registrySuffix)) {
-        continue;
-      }
-      String inferredCredentialHelperSuffix = COMMON_CREDENTIAL_HELPERS.get(registrySuffix);
-      if (inferredCredentialHelperSuffix == null) {
-        throw new IllegalStateException("No COMMON_CREDENTIAL_HELPERS should be null");
-      }
-      inferredCredentialHelperSuffixes.add(inferredCredentialHelperSuffix);
-    }
+  public CredentialRetriever wellKnownCredentialHelper() {
+    List<String> wellKnownCredentialHelpers =
+        WELL_KNOWN_CREDENTIAL_HELPERS
+            .keySet()
+            .stream()
+            .filter(imageReference.getRegistry()::endsWith)
+            .map(key -> Verify.verifyNotNull(WELL_KNOWN_CREDENTIAL_HELPERS.get(key)))
```

Comment:
This code isn't necessary as Guava's [immutable collections are "null hostile"](https://github.com/google/guava/wiki/LivingWithNullHostileCollections).